wasm2c/wasm-rt.h

```diff
@@ -313,11 +322,14 @@ uint32_t wasm_rt_exception_size(void);
 void* wasm_rt_exception(void);
 
 #if WASM_RT_MEMCHECK_SIGNAL_HANDLER && !defined(_WIN32)
-#define WASM_RT_SETJMP(buf) sigsetjmp(buf, 1)
+#define WASM_RT_SETJMP_SETBUF(buf) sigsetjmp(buf, 1)
 #else
-#define WASM_RT_SETJMP(buf) setjmp(buf)
+#define WASM_RT_SETJMP_SETBUF(buf) setjmp(buf)
 #endif
 
+#define WASM_RT_SETJMP(buf) \
+  ((buf).initialized = true, WASM_RT_SETJMP_SETBUF((buf).buffer))
```

Comment:
I've not familiar with this comma seperated syntax in C .. is it part of C89?  Would this be more clear if it was two separate statements withing a `do/while(0)`?

Reply:
The comma separate syntax is pretty convenient, as it treats several statements like a single expression and returns the last statements' value. The nice bit is that it allows use to continue to use WASM_RT_SETJMP in if statements `if (WASM_RT_SETJMP(buf) {...}`. In contrast, a `do{} while(0)` impl would not work in the prior example. Fwiw, the comma syntax is already present in wasm2c runtime --- See [here](https://github.com/WebAssembly/wabt/blob/main/wasm2c/wasm-rt-impl.h#L64-L66)

Reply:
Sounds good.